src/pages/Settings/components/NameInput/index.tsx

```diff
@@ -60,66 +66,79 @@ export const NameInput: React.FC<Props> = memo(
-
-        const data: GardenUpdateParams["data"] = {
-          name: currentGardenName,
-          fireBaseUserId: garden.fireBaseUserId,
-          gardenCategoryId: garden.gardenCategoryId,
-        };
         updateGardenInfo({
-          gardenId: garden._id,
-          data,
+          gardenId,
+          data: { ...updateData, name: currentGardenName },
         });
       },
       // eslint-disable-next-line react-hooks/exhaustive-deps
-      [garden, currentGardenName]
+      [gardenId, currentGardenName]
     );
 
     const onCancel = useCallback(() => {
       setShowNameInput(false);
       setGardenName(initialGardenName);
     }, [initialGardenName, setGardenName, setShowNameInput]);
 
+    const onEdit = useCallback(() => {
+      setShowNameInput(true);
+    }, [setShowNameInput]);
+
     useEffect(() => {
       if (updateGardenApi.status === "succeeded") {
         setShowNameInput(false);
         setGardenName(updateGardenApi.response.name || "-");
+        updateInitVal(updateGardenApi.response.name || "-");
       }
-    }, [setGardenName, setShowNameInput, updateGardenApi]);
+    }, [setGardenName, setShowNameInput, updateGardenApi, updateInitVal]);
 
     return (
       <>
         <TextField
           label="Garden Name"
-          variant="outlined"
+          variant={showInput ? "outlined" : undefined}
           className={styles.nameInput}
           value={currentGardenName}
           onChange={onNameInputChange}
+          placeholder="Add a Garden Name"
+          InputProps={{
+            readOnly: !showInput,
+          }}
           disabled={updateGardenApi.status === "loading"}
         />
         <div className={styles.nameInputButtons}>
-          <Button
-            variant="outlined"
-            onClick={onCancel}
-            disabled={updateGardenApi.status === "loading"}
-          >
-            Cancel
-          </Button>
+          {showInput ? (
+            <>
+              <Button
+                variant="outlined"
+                onClick={onCancel}
+                disabled={updateGardenApi.status === "loading"}
+              >
+                Cancel
+              </Button>
 
-          <div className={classes.buttonWrapper}>
-            <Button
-              color="primary"
-              variant="contained"
-              onClick={updateGarden}
-              disabled={disableUpdate}
-            >
-              Update
+              <div className={classes.buttonWrapper}>
+                <Button
+                  color="primary"
+                  variant="contained"
+                  onClick={updateGarden}
+                  disabled={disableUpdate}
+                >
+                  Update
+                </Button>
+                {updateGardenApi.status === "loading" && (
+                  <CircularProgress
+                    size={28}
+                    className={classes.buttonProgress}
+                  />
+                )}
```

Comment:
Right now theres a lot of similar code in each component I'd like to refactor later if we have time